classes/migration/install/RolesAndUserGroupsMigration.php

```diff
@@ -27,7 +28,8 @@ public function up(): void
         Schema::create('user_groups', function (Blueprint $table) {
             $table->comment('All defined user roles in a context, such as Author, Reviewer, Section Editor and Journal Manager.');
             $table->bigInteger('user_group_id')->autoIncrement();
-            $table->bigInteger('context_id');
+            $table->bigInteger('context_id')->nullable();
+            $table->foreign('context_id', 'user_groups_context_id')->references(Application::getContextDAO()->primaryKeyColumn)->on(Application::getContextDAO()->tableName)->onDelete('cascade');
```

Comment:
I didn't want to use application code at the migration, but I also didn't want to introduce new patterns here.